### PR TITLE
Maze Goal Generator

### PR DESCRIPTION
**Pull request by e-dzia**
Added goal generator for Maze environment

### CODE REVIEW

File: gym_maze/maze.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
+import random
+
 PATH_MAPPING = 0
 WALL_MAPPING = 1
 REWARD_MAPPING = 9
 
+STATE_ONE_STEP = 0
```
Comment (khozzy) [inline gym_maze/maze.py]: What does those states mean?

File: gym_maze/maze.py [hunk 2]
```diff
@@ -174,3 +183,27 @@ def get_possible_neighbour_cords(pos_x, pos_y) -> tuple:
         nw = (pos_x - 1, pos_y - 1)
 
         return n, ne, e, se, s, sw, w, nw
+
+    def get_goal_state(self, current_x, current_y):
```
Comment (khozzy) [inline gym_maze/maze.py:182]: What is this function returning? I assume some goal perception? How to interpret this?

File: gym_maze/maze.py (hunk 1 above)
Comment (e-dzia) [reply gym_maze/maze.py]: They describe states of the goal generator. They are not really used right now, mostly for debugging or future use.

File: gym_maze/maze.py (hunk 2 above)
Comment (e-dzia) [reply gym_maze/maze.py:182]: It returns perception of the next goal state. It is used in Action Planning phase.

**Comment by khozzy** [comment]
Please get rid of non-used code (it should only contain one that is necessary to run). Also, there are no tests of that functionality.

**Review by khozzy — APPROVED** [review]

**Pull request merged**